codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -53,11 +57,14 @@
  */
 public class Http2ConnectionHandler extends ByteToMessageDecoder implements Http2LifecycleManager {
     private static final InternalLogger logger = InternalLoggerFactory.getInstance(Http2ConnectionHandler.class);
+    private static final int DEFAULT_GRACEFUL_SHUTDOWN_TIMEOUT_MILLIS = 15 * 60 * 1000; // 15 minutes
```

Comment:
15 minutes seems _fine_. Very long if you aren't aware of it though (which is my experience with the graceful shutdown: "not aware of it and getting bitten").

I would have assumed a figure more like 5-30 seconds to just let normal HTTP requests to complete. I wonder if it would be better to default to 0 so applications know what they are getting into and that essentially any number chosen here will be "wrong".

But since any number chosen here will be "wrong," I'm not really concerned. 15 minutes works.